File: 492_ConstructTheRectangle.py
import math
class Solution(object):
    def constructRectangle(self, area):
        """
        :type area: int
        :rtype: List[int]
        """
        width, res = int(math.sqrt(area)), list()
        while width != 0:
            if area % width == 0:
                res.append(area / width)
                res.append(width)
                break
            width -= 1
        return res

File: 492_ConstructTheRectangle.cpp
//37%
class Solution {
public:
    vector<int> constructRectangle(int area) {
        int width(sqrt(area));
		vector<int> res;
		while(width){
			if(area % width == 0){
				res.push_back(area / width);
				res.push_back(width);
				break;
			}
			width--;
		}
		return res;
    }
};